<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1071</width>
    <height>520</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1081</width>
      <height>471</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>43</red>
          <green>45</green>
          <blue>48</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>43</red>
          <green>45</green>
          <blue>48</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>43</red>
          <green>45</green>
          <blue>48</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>43</red>
          <green>45</green>
          <blue>48</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>43</red>
          <green>45</green>
          <blue>48</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>43</red>
          <green>45</green>
          <blue>48</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>43</red>
          <green>45</green>
          <blue>48</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>43</red>
          <green>45</green>
          <blue>48</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>43</red>
          <green>45</green>
          <blue>48</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">
background-color: rgb(43, 45, 48);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Raised</enum>
    </property>
    <widget class="QTextEdit" name="inputCode">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>471</width>
       <height>271</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>10</pointsize>
       <italic>false</italic>
       <bold>false</bold>
      </font>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(0, 0, 0);
font: 10pt &quot;Segoe UI&quot;;
color: rgb(255, 255, 255);</string>
     </property>
     <property name="html">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>570</x>
       <y>10</y>
       <width>151</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 255, 255);
background-color: rgb(43, 45, 48);
font: 700 12pt &quot;Segoe UI&quot;;</string>
     </property>
     <property name="text">
      <string>Parallel Version</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>570</x>
       <y>260</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 255, 255);
background-color: rgb(43, 45, 48);
font: 700 12pt &quot;Segoe UI&quot;;</string>
     </property>
     <property name="text">
      <string>Output</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="parallelCode">
     <property name="geometry">
      <rect>
       <x>570</x>
       <y>50</y>
       <width>481</width>
       <height>191</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>10</pointsize>
       <italic>false</italic>
       <bold>false</bold>
      </font>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(0, 0, 0);
font: 10pt &quot;Segoe UI&quot;;
color: rgb(255, 255, 255);</string>
     </property>
     <property name="html">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="outputLog">
     <property name="geometry">
      <rect>
       <x>570</x>
       <y>290</y>
       <width>481</width>
       <height>151</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>10</pointsize>
       <italic>false</italic>
       <bold>false</bold>
      </font>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(0, 0, 0);
font: 10pt &quot;Segoe UI&quot;;
color: rgb(255, 255, 255);</string>
     </property>
     <property name="html">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="procCount">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>390</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">/* 1. Style the Main Box */
QSpinBox {
    color: white;
    background-color: #333333;   /* Dark background */
    border: 2px solid #555555;   /* Gray border */
    border-radius: 4px;          /* Rounded corners */
    padding-right: 20px;         /* Make room for the buttons on the right */
    selection-background-color: #0078d7;
}

/* 2. Style the UP Button */
QSpinBox::up-button {
    subcontrol-origin: border;
    subcontrol-position: top right; /* Pin to top-right */
    width: 20px;                    /* Width of the button */
    
    background: #555555;            /* Button Color */
    border-left: 1px solid #333333; /* Separator line */
    border-bottom: 1px solid #333333;
    border-top-right-radius: 4px;
}

/* 3. Style the DOWN Button */
QSpinBox::down-button {
    subcontrol-origin: border;
    subcontrol-position: bottom right; /* Pin to bottom-right */
    width: 20px;
    
    background: #555555;
    border-left: 1px solid #333333;
    border-top: 1px solid #333333;
    border-bottom-right-radius: 4px;
}

/* 4. Add Hover Effects (So you know it's clickable) */
QSpinBox::up-button:hover, QSpinBox::down-button:hover {
    background: #777777;
}

QSpinBox::up-button:pressed, QSpinBox::down-button:pressed {
    background: #999999;
}

/* 5. Force Arrows to be visible (Optional, depending on your system) */
QSpinBox::up-arrow, QSpinBox::down-arrow {
    width: 7px;
    height: 7px;
    background: none; 
    border-left: 1px solid white;  /* Draw a simple arrow using borders */
    border-top: 1px solid white;   /* (This creates a 'V' shape) */
}

QSpinBox::up-arrow { 
    image: none; 
    transform: rotate(-45deg);     /* Point up */
    margin-top: 1px;
}

QSpinBox::down-arrow { 
    image: none; 
    transform: rotate(135deg);     /* Point down */
    margin-bottom: 1px;
}</string>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="value">
      <number>4</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>350</y>
       <width>381</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 255, 255);
background-color: rgb(43, 45, 48);
font: 700 12pt &quot;Segoe UI&quot;;</string>
     </property>
     <property name="text">
      <string>Select Number of Processors to run on</string>
     </property>
    </widget>
    <widget class="QPushButton" name="runButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>93</width>
       <height>29</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>-1</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton {
    /* VS Dark Gray Background */
    background-color: rgb(43, 45, 48);
    
    /* VS Green for the text/icon */
    color: #5db55b;  
    
    /* Subtle border to match the IDE look */
    border: rgb(43, 45, 48);
    border-radius: 2px;
    
    /* Spacing and Font */
    padding: 5px 15px;
    font-family: &quot;Segoe UI&quot;, sans-serif;
    font-size: 14px;
    font-weight: bold;
}

QPushButton:hover {
    /* Lighter gray when hovering */
    background-color: #3e3e40;
    border-color: #999999;
}

QPushButton:pressed {
    /* Darker or Blue-ish accent when clicked */
    background-color: #007acc;
    color: white;
    border-color: #007acc;
}</string>
     </property>
     <property name="text">
      <string>â–¶ Run</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1071</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
